### PR TITLE
User update profile

### PR DESCRIPTION
**Pull request by shruska**
### Add AuthContext.tsx

  **Context Creation:** 
  AuthContext is created to hold the authentication state.
  
  **Provider Component:**
   AuthProvider maintains the authentication state and provides it to the component tree.
  
  **Custom Hook:**
   useAuth simplifies accessing authentication state and actions in functional components.

### Add ProfileEdit.tsx page
update email and password

###  Add Routes.tsx
centralized link management

### CODE REVIEW

File: Pet-Clinic-Frontend/src/main.tsx [hunk 1]
```diff
@@ -3,29 +3,58 @@ import ReactDOM from 'react-dom/client'
...
+        element: <Login />,
+      },
+      {
+        path: routes[3].path,
+        element: (
+         <PrivateRoute>
+            <ProfileEdit />
+        </PrivateRoute>
+        ),
+      },
```
Comment (markkovari) [inline Pet-Clinic-Frontend/src/main.tsx:49]:
```suggestion
      ...routes.map(({ path, requiresAuth }) => ({
        path,
        element: requiresAuth ? (
          <PrivateRoute>
            <ProfileEdit />
          </PrivateRoute>
        ) : <ProfileEdit />,
      })),
```

Comment (markkovari) [reply Pet-Clinic-Frontend/src/main.tsx:49]: You can make create the router dynamically too.

File: Pet-Clinic-Frontend/src/pages/Landing.tsx [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
 import clinicImage from "../images/petclinic.png";
 
-function PageView() {
+const Landing = () => {
+
   return (
-    <>
-   
-    <img src={clinicImage} alt="Clinic" className="clinic-image" /> 
+    <div>
```
Comment (markkovari) [inline Pet-Clinic-Frontend/src/pages/Landing.tsx:7]: The react fragment should be sufficient here.

File: Pet-Clinic-Frontend/src/pages/Login.tsx [hunk 3]
```diff
@@ -36,9 +38,10 @@ const Login = () => {
       });
       if (!response.status.toString().startsWith('2')) {
```
Comment (markkovari) [inline Pet-Clinic-Frontend/src/pages/Login.tsx:39]:
```suggestion
      if (!response.status.toString().startsWith('2')) {
```
I think we should not make this number a string to check if it is 2xx, we can check it with numbers too.

File: Pet-Clinic-Frontend/src/pages/Login.tsx [hunk 4]
```diff
@@ -36,9 +38,10 @@ const Login = () => {
...
         setError('Invalid email or password');
+      } else {
+        login()
```
Comment (markkovari) [inline Pet-Clinic-Frontend/src/pages/Login.tsx:42]: Well this login is a bit counter intuitive for me. Shouldn't we store the user somehow here? Maybe it is not implemented yet, but we should not just "say" that we are logged in. We should store something at least.

File: Pet-Clinic-Frontend/src/pages/ProfileEdit.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    try {
+      const response = await fetch(`${backendUrl}/profile-edit`, { // what endpoint is in backEnd??
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ email, password }),
+      });
+
+      if (!response.status.toString().startsWith('2')) {
```
Comment (markkovari) [inline Pet-Clinic-Frontend/src/pages/ProfileEdit.tsx:24]: Same here with the startswith

File: Pet-Clinic-Frontend/src/pages/ProfileEdit.tsx [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        setError('Failed to update profile. Please try again.');
+      } else {
+        setSuccess('Profile updated successfully!');
+      }
+
+    } catch (error) {
+      console.error('Error:', error);
+      setError('An error occurred. Please try again later.');
+    }
+  };
```
Comment (markkovari) [inline Pet-Clinic-Frontend/src/pages/ProfileEdit.tsx:34]: This function as is might be in a different module too 😉

File: Pet-Clinic-Frontend/src/routes.tsx [hunk 7]
```diff
@@ -0,0 +1,6 @@
+export const routes = [
```
Comment (markkovari) [inline Pet-Clinic-Frontend/src/routes.tsx:1]: I think we should rely on some typing here to avoid human mistakes.

**Review by markkovari — CHANGES_REQUESTED** [review]

File: Pet-Clinic-Frontend/src/main.tsx (hunk 1 above)
Comment (shruska) [reply Pet-Clinic-Frontend/src/main.tsx:49]: New code always ends in <ProfileEdit>.  It does not seem to me beneficial :-).  OK I understand  I will find out how to provide different component/pages.

Comment (shruska) [reply Pet-Clinic-Frontend/src/main.tsx:49]: always ends in ProfileEdit

File: Pet-Clinic-Frontend/src/pages/Landing.tsx (hunk 2 above)
Comment (shruska) [reply Pet-Clinic-Frontend/src/pages/Landing.tsx:7]: Ok. It is only about code cleanliness?  <Div> is redundant?

File: Pet-Clinic-Frontend/src/pages/Login.tsx (hunk 3 above)
Comment (shruska) [reply Pet-Clinic-Frontend/src/pages/Login.tsx:39]: If i use for example (response.status < 200 || response.status >= 300).  Why is better, it is better?

File: Pet-Clinic-Frontend/src/pages/Login.tsx (hunk 4 above)
Comment (shruska) [reply Pet-Clinic-Frontend/src/pages/Login.tsx:42]: I don't know, maybe. It is ok for me at least now.  I would rather discuss this in class channel tomorrow.

File: Pet-Clinic-Frontend/src/pages/ProfileEdit.tsx (hunk 5 above)
Comment (shruska) [reply Pet-Clinic-Frontend/src/pages/ProfileEdit.tsx:24]: ok

File: Pet-Clinic-Frontend/src/pages/ProfileEdit.tsx (hunk 6 above)
Comment (shruska) [reply Pet-Clinic-Frontend/src/pages/ProfileEdit.tsx:34]: yes it is obvious. :-)

File: Pet-Clinic-Frontend/src/routes.tsx (hunk 7 above)
Comment (shruska) [reply Pet-Clinic-Frontend/src/routes.tsx:1]: I don't understand what is wrong? the word  " routes "?

File: Pet-Clinic-Frontend/src/pages/Landing.tsx (hunk 2 above)
Comment (markkovari) [reply Pet-Clinic-Frontend/src/pages/Landing.tsx]: well it is rather about "semantic html", we rather should not inject more divs then necessary

File: Pet-Clinic-Frontend/src/pages/Login.tsx (hunk 3 above)
Comment (markkovari) [reply Pet-Clinic-Frontend/src/pages/Login.tsx:39]: Well status codes are numbers and we should treat them as such, I am not saying this does not work, but it is debatable, which one is more understandable.

File: Pet-Clinic-Frontend/src/main.tsx (hunk 1 above)
Comment (markkovari) [reply Pet-Clinic-Frontend/src/main.tsx]: Oh, yes you are correct, you should have a reference to the components too, which one should be rendered. This was just an example. THe type that you wanted to recreate is the https://reactrouter.com/en/main/routers/create-browser-router#type-declaration
type declaration from here.

File: Pet-Clinic-Frontend/src/routes.tsx (hunk 7 above)
Comment (markkovari) [reply Pet-Clinic-Frontend/src/routes.tsx]: No, not really. Nothing is preventing you to have a typo or duplicated fields and so on. If we are using typescript, we might wanted to use the types too.

**Review by markkovari — APPROVED** [review]

**Review by Metemi — APPROVED** [review]

**Pull request merged**